File: 2868-continuous-subarrays/continuous-subarrays.cpp
class Solution {
public:
    long long continuousSubarrays(vector<int>& a) {
        int n = (int)a.size();
        long long ans = 0;
        int l = 0;
        multiset<int>s;
        for(int i = 0 ; i < n ; i++) {
            s.insert(a[i]);
            while(abs(*s.rbegin() - *s.begin()) > 2) {
                s.erase(s.find(a[l]));
                l++;
            }
            ans += (i-l+1);
        }
        return ans;
    }
};

File: 2868-continuous-subarrays/continuous-subarrays.py
class Solution:
    def continuousSubarrays(self, nums: List[int]) -> int:
        left = 0
        freq_map = {}
        count = 0
        
        for right in range(len(nums)):
            # Increment frequency of nums[right] in freq_map
            freq_map[nums[right]] = freq_map.get(nums[right], 0) + 1
            
            # Find the maximum and minimum values in the current window
            max_val = max(freq_map)
            min_val = min(freq_map)
            
            # Check if conditions are satisfied
            while max_val - min_val > 2:
                freq_map[nums[left]] -= 1
                if freq_map[nums[left]] == 0:
                    del freq_map[nums[left]]
                left += 1
                max_val = max(freq_map)
                min_val = min(freq_map)
            
            # Increment count
            count += right - left + 1
        
        return count
